packages/instantsearch.js/src/middlewares/createInsightsMiddleware.ts

```diff
@@ -230,7 +224,42 @@ export function createInsightsMiddleware<
           if (onEvent) {
             onEvent(event, _insightsClient as TInsightsClient);
           } else if (event.insightsMethod) {
-            insightsClient(event.insightsMethod, event.payload);
+            // We keep the queue length for later.
+            const queueLength = insightsClient.queue?.length || 0;
+
+            try {
+              // We first try to send the event.
+              insightsClient(event.insightsMethod, event.payload);
+            } catch (e) {
+              // We catch if `search-insights` is not initialized yet
+              // and we `init` before sending the event.
+              insightsClient('init', {
+                appId,
+                apiKey,
+                useCookie: true,
```

Comment:
maybe these should be factored inside insightsInitParams?

Reply:
Is that what you were thinking? https://github.com/algolia/instantsearch/pull/5523/commits/531ce681ba3b4336f916b61f4f66e4859bf8d707